reference-implementation/test/readable-byte-stream.js

```diff
@@ -0,0 +1,1415 @@
+      t.end();
+    }
+  });
+
+  const reader = rbs.getReader();
+
+  controller.enqueue(new Uint8Array(16));
+  controller.close();
+
+  reader.read().then(result => {
+    t.equals(result.done, false, 'done');
+
+    const view = result.value;
+    t.equals(view.byteOffset, 0, 'byteOffset');
+    t.equals(view.byteLength, 16, 'byteLength');
+
+    return reader.read();
+  }).then(result => {
+    t.equals(result.done, true, 'done');
+    t.equals(result.value, undefined, 'value');
+
+    t.end();
+  }).catch(e => {
+    t.fail(e);
+    t.end();
+  });
+});
+
+test('ReadableByteStream: enqueue(), close(), getReader(), then read()', t => {
+  const rbs = new ReadableByteStream({
+    start(c) {
+      c.enqueue(new Uint8Array(16));
+      c.close();
+    },
+    pull() {
+      t.fail('pull must not be called');
+      t.end();
+    },
+    pullInto() {
+      t.fail('pullInto must not be called');
+      t.end();
+    }
+  });
+
+  const reader = rbs.getReader();
+
+  reader.read().then(result => {
+    t.equals(result.done, false, 'done');
+
+    const view = result.value;
+    t.equals(view.byteOffset, 0, 'byteOffset');
+    t.equals(view.byteLength, 16, 'byteLength');
+
+    return reader.read();
+  }).then(result => {
+    t.equals(result.done, true, 'done');
+    t.equals(result.value, undefined, 'value');
+
+    t.end();
+  }).catch(e => {
+    t.fail(e);
+    t.end();
+  });
+});
+
+test('ReadableByteStream: read(), then enqueue()', t => {
+  let controller;
+
+  const rbs = new ReadableByteStream({
+    start(c) {
+      controller = c;
+    },
+    pull() {
+      controller.enqueue(new Uint8Array(16));
+    },
+    pullInto() {
+      t.fail('pullInto must not be called');
+      t.end();
+    }
+  });
+
+  const reader = rbs.getReader();
+
+  reader.read().then(result => {
+    t.equals(result.done, false, 'done');
+    t.equals(result.value.byteLength, 16, 'byteLength');
+
+    t.end();
+  }).catch(e => {
+    t.fail(e);
+    t.end();
+  });
+});
+
+test('ReadableByteStream: read(), then respond()', t => {
+  let controller;
+
+  const rbs = new ReadableByteStream({
+    start(c) {
+      controller = c;
+    },
+    pull() {
+      try {
+        controller.respond(1);
+      } catch(e) {
+        t.equals(e.constructor, TypeError);
+
+        t.end();
+        return;
+      }
+
+      t.fail('respond() didn\'t throw');
+      t.end();
+    },
+    pullInto() {
+      t.fail('pullInto must not be called');
+      t.end();
+    }
+  });
+
+  const reader = rbs.getReader();
+
+  reader.read().catch(e => {
+    t.fail(e);
+    t.end();
+  });
+});
+
+test('ReadableByteStream: read(view), then respond() with too big value', t => {
+  let controller;
+
+  const rbs = new ReadableByteStream({
+    start(c) {
+      controller = c;
+    },
+    pull() {
+      t.fail('pull must not be called');
+      t.end();
+    },
+    pullInto(buffer, offset, length) {
+      try {
+        controller.respond(2);
```

Comment:
Again, including both arguments is important to make it clearer what the bad behavior is that's causing the error.